Assets/Mirror/Runtime/Transport/Transport.cs

```diff
@@ -1,41 +1,43 @@
-﻿// transport layer backend
-// - set to telepathy by default
-// - can be changed by assigning Transport.layer to whatever you want
+﻿// abstract transport layer component
+using UnityEngine;
+
 namespace Mirror
 {
-    // Transport class used by HLAPI ///////////////////////////////////////////
-    public static class Transport
-    {
-        // selected transport layer
-        // the transport is normally initialized in NetworkManager InitializeTransport
-        // initialize it yourself if you are not using NetworkManager
-        public static TransportLayer layer;
-    }
-
-    // abstract transport layer class //////////////////////////////////////////
     // note: 'address' is ip / websocket url / ...
     public enum TransportEvent { Connected, Data, Disconnected }
-    public interface TransportLayer
+
+    public abstract class Transport : MonoBehaviour
```

Comment:
Make this implement "ITransport" instead of having all these abstract methods.

Gives me the flexibility to have a non component based transport,  which I will need for multiplex transport